components/Modals/ObtainItemModal.tsx

```diff
@@ -0,0 +1,522 @@
+    return false;
+  }, [selectedItem]);
+
+  const isWhitelistedByMerkleProof = useMemo(() => {
+    if (!selectedItem) return false;
+    if (selectedItem.merkleRoot === pad('0x00')) return false;
+    return whitelistedAmount > BigInt(0);
+  }, [selectedItem, whitelistedAmount]);
+
+  const { isApprovedForAll } = useIsApprovedForAll(
+    game?.itemsAddress,
+    character?.account,
+    game?.itemsManager,
+  );
+
+  const satisfiesClaimRequirements = useMemo(() => {
+    if (!selectedItem) return false;
+    if (!character) return false;
+    if (!game) return false;
+    return checkClaimRequirements(
+      selectedItem.claimRequirements,
+      game,
+      character,
+    );
+  }, [selectedItem, character, game]);
+
+  const satisfiesCraftRequirements = useMemo(() => {
+    if (!selectedItem) return false;
+    if (!character) return false;
+    if (!amount) return false;
+    return checkCraftRequirements(
+      selectedItem.craftRequirements,
+      character,
+      BigInt(amount),
+    );
+  }, [selectedItem, character, amount]);
+
+  const onObtainItem = useCallback(async () => {
+    if (noSupply) {
+      throw new Error('This item has zero supply.');
+    }
+
+    if (noDistribution) {
+      throw new Error(
+        'You have already claimed the maximum amount of this item.',
+      );
+    }
+
+    if (hasError && isWhitelistedForAll) {
+      setShowError(true);
+      return null;
+    }
+
+    if (!walletClient) {
+      throw new Error('Could not find a wallet client');
+    }
+
+    if (!game?.itemsAddress) {
+      throw new Error('Missing game data');
+    }
+
+    if (!character) {
+      throw new Error('Character address not found');
+    }
+
+    if (!selectedItem) {
+      throw new Error('Item not found');
+    }
+
+    setIsObtaining(true);
+
+    if (!isWhitelistedForAll && !tree) {
+      console.error('Could not find the whitelist tree.');
+      throw new Error(
+        `Something went wrong while obtaining ${selectedItem.name}.`,
+      );
+    }
+
+    const itemId = BigInt(selectedItem.itemId);
+
+    let proof: string[] = [];
+    let obtainingAmount = BigInt(amount);
+
+    if (
+      isWhitelistedByMerkleProof &&
+      tree &&
+      whitelistedAmount > BigInt(0) &&
+      whitelistLeaf
+    ) {
+      proof = tree.getProof(whitelistLeaf);
+      obtainingAmount = whitelistedAmount;
+    } else if (!isWhitelistedForAll) {
+      console.error('Not whitelist by public or merkle proof.');
+      throw new Error(
+        `Something went wrong while obtaining ${selectedItem.name}.`,
+      );
+    }
+
+    if (isCraftable && !satisfiesCraftRequirements) {
+      throw new Error('You do not have the required items to craft this item.');
+    }
+
+    if (
+      !isCraftable &&
+      !!selectedItem?.claimRequirements &&
+      !satisfiesClaimRequirements
+    ) {
+      throw new Error('You do not meet the requirements to claim this item.');
+    }
+
+    try {
```

Comment:
I think this is a relic of a mistake I made with the original modal, but `try` should be moved to the beginning of the handler

Reply:
Fixed in 554d784d0b4baa331b23a5ce7becf2299cacce73